packages/asset-service/src/generateAssetData/generateAssetData.ts

```diff
@@ -33,36 +35,60 @@ const generateAssetData = async () => {
     ...osmosisAssets,
     ...avalancheAssets,
     ...optimismAssets,
+    ...bnbsmartchainAssets,
   ]
+
   // remove blacklisted assets
   const filteredAssetData = filterOutBlacklistedAssets(unfilteredAssetData)
 
-  // For coins not currently in the color map, check to see if we can generate a color from the icon
-  const filteredWithColors = await setColors(filteredAssetData)
-
   // deterministic order so diffs are readable
-  const orderedAssetList = orderBy(filteredWithColors, 'assetId')
+  const orderedAssetList = orderBy(filteredAssetData, 'assetId')
+
+  const evmAssetNamesByChainId = {
+    [KnownChainIds.EthereumMainnet]: ethAssets.map((asset) => asset.name),
+    [KnownChainIds.AvalancheMainnet]: avalancheAssets.map((asset) => asset.name),
+    [KnownChainIds.OptimismMainnet]: optimismAssets.map((asset) => asset.name),
+    [KnownChainIds.BnbSmartChainMainnet]: bnbsmartchainAssets.map((asset) => asset.name),
+  }
 
-  const ethAssetNames = ethAssets.map((asset) => asset.name)
-  const avalancheAssetNames = avalancheAssets.map((asset) => asset.name)
-  const optimismAssetNames = optimismAssets.map((asset) => asset.name)
+  const isNotUniqueAsset = (asset: Asset) => {
```

Comment:
Since that's really encapsulated into `generateAssetData` so not a big deal, but I guess something something multi-chain would be a better terminology

Reply:
going to leave it as is for now, easy to update if causes confusion in future, but want to get this in